src/batch-transaction.ts

```diff
@@ -126,20 +137,34 @@ class BatchTransaction extends Snapshot {
    * @example <caption>include:samples/batch.js</caption>
    * region_tag:spanner_batch_client
    */
-  createQueryPartitions(query, callback) {
+  createQueryPartitions(
+    query: string | ExecuteSqlRequest,
+  ): Promise<CreateQueryPartitionsResponse>;
+  createQueryPartitions(
+    query: string | ExecuteSqlRequest,
+    callback: CreateQueryPartitionsCallback,
+  ): void;
+  createQueryPartitions(
+    query: string | ExecuteSqlRequest,
+    cb?: CreateQueryPartitionsCallback,
+  ): void | Promise<CreateQueryPartitionsResponse> {
     if (is.string(query)) {
```

Comment:
instead use 

```
const request: ExecuteSqlRequest =
  typeof query === 'string' ? { sql: query } : query;
```

Reply:
You can use any other variable name if required. In the current implementation we are doing the type cast for every reference `query as ExecuteSqlRequest`

Reply:
thanks for the suggestion, I have updated the code!